storage/src/nodestore.rs

```diff
@@ -694,14 +694,34 @@ impl Version {
     const SIZE: u64 = std::mem::size_of::<Self>() as u64;
 
     /// construct a [Version] header from the firewood version
-    fn new() -> Self {
-        let mut version_bytes: [u8; Self::SIZE as usize] = Default::default();
-        let version = env!("CARGO_PKG_VERSION");
-        let _ = version_bytes
-            .as_mut_slice()
-            .write_all(format!("firewood {}", version).as_bytes());
+    const fn new() -> Self {
+        // statically construct the version string using the env variable set by build.rs
+        const VERSION_STR: &str = concat!("firewood ", env!("FIREWOOD_DB_VERSION"));
+
+        // compile-time assertion that the version string is not too long.
+        const {
+            assert!(
+                VERSION_STR.len() <= Self::SIZE as usize,
+                "Version string too long"
+            );
+        }
+
+        // statically extend the version string to the full size of the Version struct
```

Comment:
format strings are magic, and can be fixed size with padding. Suggest using format! to get this string padded with the bytes you want.